studio/components/layouts/ProjectLayout/LayoutHeader/NotificationsPopover/NotificationActions.tsx

```diff
@@ -1,39 +1,104 @@
 
   const onSelectUpgradeProject = () => {
     return router.push(`/project/${project.ref}/settings/billing/update/pro`)
   }
 
+  // TODO: allow configuring other routes
+  const onSelectMigrateProject = () => {
+    post(`${API_URL}/platform/database/${project.ref}/owner-reassign`, {})
+      // Refresh database connection string
+      .then((_) => app.projects.fetchDetail(project.ref))
+      .catch((err) => console.info(err))
+  }
+  const onSelectRollbackProject = () => {
+    delete_(`${API_URL}/platform/database/${project.ref}/owner-reassign`, {})
+      // Refresh database connection string
+      .then((_) => app.projects.fetchDetail(project.ref))
+      .catch((err) => console.info(err))
+  }
+  const onSelectFinaliseProject = () => {
+    patch(`${API_URL}/platform/database/${project.ref}/owner-reassign`, {})
+      // Refresh database connection string
+      .then((_) => app.projects.fetchDetail(project.ref))
+      .catch((err) => console.info(err))
+  }
+
   return (
     <div className="space-y-2">
-      {availableActions.map((action: any) => {
-        if (action.action_type === 'project.upgrade') {
+      {availableActions.map((action) => {
+        if (action.action_type === ActionType.UpgradeProjectToPro) {
           return (
             <Button key={action.action_type} type="default" onClick={onSelectUpgradeProject}>
               Upgrade project
             </Button>
           )
-        } else if (action.action_type === 'postgresql.restart') {
+        } else if (action.action_type === ActionType.SchedulePostgresRestart) {
           return (
             <Button key={action.action_type} type="default" onClick={onSelectRestartProject}>
               Restart project
             </Button>
           )
+        } else if (action.action_type === ActionType.MigratePostgresSchema) {
+          if (action.reason === 'finalise') {
+            return (
+              <Button
+                key={`${action.action_type}_${action.reason}`}
+                type="default"
+                onClick={() => setShowModal(true)}
+              >
+                Finalise
+              </Button>
+            )
+          }
+          if (action.reason === 'rollback') {
+            return (
+              <Button
+                key={`${action.action_type}_${action.reason}`}
+                type="default"
+                onClick={onSelectRollbackProject}
+              >
+                Rollback
+              </Button>
+            )
+          }
+          return (
+            <Button key={action.action_type} type="default" onClick={onSelectMigrateProject}>
+              Apply now
+            </Button>
+          )
         }
       })}
+      <ConfirmModal
+        danger
+        visible={showModal}
+        title={`Schema migration for "${project.name}"`}
+        description={`Are you sure you want to reassign entities owned by temporary role to postgres? This action is irreversible.`}
```

Comment:
this seems very specific to this particular migration

Reply:
Thanks for catching this issue. Update screenshot in PR